### ISSUE TITLE
Player: The video finishes without paella noticing.

### ISSUE THREAD

**Issue by Alfro** [polimediaupv/paella#213]
For certain video lengths, the pause event is not thrown and the UI does not update to display the end of the video and the play button)

_Screen capture below:_
![screen capture](https://cloud.githubusercontent.com/assets/11334172/16811057/e1b5a872-4926-11e6-8bb4-e6dd51a36740.gif)

This is true for 4.2.x, I sadly couldn't make sure about 5.0.x:
- First, the 'currentTime' of a video never seems to have the 'videoLength' value, it is always less.
- Then, when `checkVideoTrimming`(on 04_video_container.js) is called the last time, 'currentTime' will have a value around 250 miliseconds smaller than the actual duration of the video.

If the duration of a video is close to an exact integer (like 8 seconds), this code would not execute:

``` js
checkVideoTrimming:function() {
    var current = this.currentTime();
    var end = this.duration();
    ...
    if (current>=Math.floor(end) && !this.paused()) {
        ...
        //These will not be called for end ~ 8 (because current is ~7.95)
        this.pause();
        paella.events.trigger(paella.events.endVideo,{videoContainer:thisClass});
        ...
```

As a workaround for 4.2.x, I dropped the `Math.floor(end)` statement and just did this instead: https://github.com/teltek/paella/commit/80fbe43f6c0bde6afc114018d9b212b317ec65be

I didn't PR because I wanted to check first whether I am right about the root of the problem. Also, I could not test it for paella 5.0.x, but I feel the issue would be the same.

**Comment by ferserc1** [comment]
This problem does not occurs in Paella 5 because all the low level video playback and synchronization mechanisms were drastically improved, thus this bug (and probably many others) were fixed.

If you think that this fix works, you can submit a pull request to old/4.3.x branch.

The only reason to keep the Paella 4.3 branch is that in Paella 5 the editor has been removed, and we are creating a new one in an independent repository, but currently we are about to release the first beta of Paella Editor.

If you are using the editor, I suggest you to test the new one at https://github.com/polimediaupv/paella-editor, and migrate to Paella 5 that fixes a lot of problems such as this one.

**Comment by rubenrua** [comment]
We are looking forward to test the new editor. Let us know when the beta comes out.

**Comment by miesgre** [comment]
@Alfro @rubenrua Is this bug fixed in paella 5.1?

**Comment by rubenrua** [comment]
We didn't reproduce the bug in paella 5.1. Close this issue and if it fails again we will create a new issue.
